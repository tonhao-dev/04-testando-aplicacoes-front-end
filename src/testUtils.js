import { render, waitFor } from '@testing-library/react';

// Providers
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from './providers/theme';
import Providers from './providers';

// Utils
export const AnyPage = () => <div data-testid="any page" />
export const waitTimersByTime = async time => {
  await waitFor(() => {
    jest.advanceTimersByTime(time);
  });
};

// Renders
export const renderWithTheme = (ui, options) => render(ui, { wrapper: ThemeProvider, ...options });
export const renderWithRouter = (ui, options) => render(ui, { wrapper: BrowserRouter, ...options });
export const renderWithProviders = (ui, options) => render(ui, { wrapper: Providers, ...options });
