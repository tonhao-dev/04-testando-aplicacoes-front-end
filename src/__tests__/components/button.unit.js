// Utilitários
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { renderWithTheme } from '../../testUtils';

// Componente
import Button from '../../components/button';

describe('<Button />', () => {
  it('Renderiza um botão corretamente', () => {
    renderWithTheme(<Button>conteudo</Button>);
    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });

  it('Executa ações de click ao receber a função por prop', () => {
    const onClick = jest.fn();
    renderWithTheme(<Button onClick={onClick}>conteudo</Button>);
    const button = screen.getByRole('button');

    userEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('Pode renderizar um ícone', () => {
    const icon = <span>icone</span>;
    renderWithTheme(<Button icon={icon}>conteudo</Button>);

    const icone = screen.getByText('icone');
    expect(icone).toBeInTheDocument();
  });
});
