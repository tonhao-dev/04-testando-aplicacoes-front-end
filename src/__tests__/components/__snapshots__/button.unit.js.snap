// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Renderiza um bot√£o corretamente 1`] = `
<button
  class="sc-hKgILt iQgQJK"
  type=""
>
  <span>
    conteudo
  </span>
</button>
`;
